Account/FxADevice.swift

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public struct FxADevice {
+    let name: String
+    let id: String?
+    let type: String?
+
+    private init(name: String, id: String?, type: String?) {
+        self.name = name
+        self.id = id
+        self.type = type
+    }
+
+    static func forRegister(name: String, type: String) -> FxADevice {
+        return FxADevice(name: name, id: nil, type: type)
+    }
+
+    static func forUpdate(name: String, id: String) -> FxADevice {
+        return FxADevice(name: name, id: id, type: nil)
+    }
+
+    func toJSON() -> JSON {
+        var parameters = [String: String]()
+        parameters["name"] = name
+        parameters["id"] = id
+        parameters["type"] = type
+
+        return JSON(parameters)
+    }
+
+    static func fromJSON(json: JSON) -> FxADevice? {
+        guard !json.isError,
```

Comment:
Nit: Looking back at the some of the code added to the FxAClient10 class, I wonder if we can just fold the isError check into the guard statements instead of having a separate if statement?